lib/engine/game/g_1880/game.rb

```diff
@@ -323,25 +329,70 @@ def init_minors
           game_minors.map { |minor| G1880::Minor.new(**minor) }
         end
 
+        def setup
+          @fully_funded_corporations = []
+          @full_cap_event = false
+          @foreign_investors_can_build = true
+          setup_building_permits
+          setup_unsaleable_shares
+        end
+
         def p1
           company_by_id('P1')
         end
 
         def status_array(corporation)
-          return [] unless corporation.floated?
+          return unless corporation.floated?
+          return if corporation.minor?
 
           ["Building Permits: #{corporation.building_permits}"]
         end
 
+        def non_floated_corporations
+          @corporations.each { |c| yield c if !c.minor? && !c.floated? }
+        end
+
         def corporation_show_individual_reserved_shares?(_corporation)
           false
         end
 
+        def event_all_shares_available!
+          @log << "-- Event: #{EVENTS_TEXT['all_shares_available'][1]} --"
+          @corporations.each do |c|
+            c.shares.last(5).each { |s| s.buyable = true }
```

Comment:
This is logically correct, but it does mean you have to keep the "magic number" 5 in multiple places. I probably would have done something like `c.shares.each { |s| s.buyable = true}` in order to remove that dependency.